src/Middleware/RateLimiting/src/RateLimitingMiddleware.cs

```diff
@@ -242,6 +255,12 @@ private PartitionedRateLimiter<HttpContext> CreateEndpointLimiter()
         }, new DefaultKeyTypeEqualityComparer());
     }
 
+    private void AddRateLimiterStatisticsFeature(HttpContext context)
+    {
+        _statisticsFeature?.SetHttpContext(context);
```

Comment:
Oh, this stores the HttpContext, we need to allocate it every time then. You can't reuse it because parallel requests will end up accessing someone else's HttpContext.

Reply:
Sorry for the extra work, my bad 😄